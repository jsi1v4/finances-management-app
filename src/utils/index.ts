import { createElement } from 'react';

/**
 * Wrap up Components
 * @param JSX.Element[] sequence of elements
 * @return wrapper (first element as father and others as children)
 */
export function wrapse(...comps) {
  return comps.reduceRight((wrap, comp) => ({ children }) =>
    createElement(wrap, null, createElement(comp, null, children))
  );
}

export default {
  wrapse
};
