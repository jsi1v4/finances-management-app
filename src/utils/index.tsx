import React from 'react';

/**
 * Wrap up Components
 * @param JSX.Element[] sequence of elements
 * @return wrapper (first element as father and others as children)
 */
export function wrapse(...components) {
  return components.reduceRight(
    (Wrapper, Comp) => ({ children }) => (
      <Wrapper>
        <Comp>{children}</Comp>
      </Wrapper>
    ),
    ({ children }) => <>{children}</>
  );
}

export default {
  wrapse
};
