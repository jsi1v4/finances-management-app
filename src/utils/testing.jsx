import React from 'react';

import { LocaleConfig } from '../i18n';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

export function LocaleTesting({ children }) {
  return <LocaleConfig lang="en">{children}</LocaleConfig>;
}

export default LocaleTesting;
