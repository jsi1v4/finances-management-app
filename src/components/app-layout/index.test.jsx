import React from 'react';
import { render } from '@testing-library/react';

import { LocaleTesting } from '../../utils/testing';
import { AppLayout } from './index';

describe('AppLayout component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<AppLayout />, {
      wrapper: LocaleTesting,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component', async () => {
    const { getByTestId, getByText } = render(<AppLayout />, {
      wrapper: LocaleTesting,
    });
    expect(getByTestId('app-layout').innerHTML).toBeDefined();
    expect(getByText('Finances Management').innerHTML).toBeDefined();
  });
});
