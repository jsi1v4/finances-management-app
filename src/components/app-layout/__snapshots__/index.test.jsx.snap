// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="ant-layout ant-layout-has-sider"
    data-testid="app-layout"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-dark"
      style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-row styles__Centered-t9ggvx-4 iLwpPw"
        >
          <img
            class="styles__Logo-t9ggvx-0 fiTTSr"
            src="https://raw.githubusercontent.com/jsi1v4/jsi1v4/master/assets/logo192.png"
          />
        </div>
      </div>
    </aside>
    <section
      class="ant-layout styles__FullPage-t9ggvx-1 gLlzLV"
    >
      <div
        class="ant-page-header styles__Header-t9ggvx-2 cCuuAs ant-page-header-ghost"
      >
        <div
          class="ant-page-header-heading"
        >
          <div
            class="ant-page-header-heading-left"
          >
            <span
              class="ant-page-header-heading-title"
            >
              Finances Management
            </span>
          </div>
        </div>
      </div>
      <section
        class="ant-layout styles__Content-t9ggvx-3 jkkToD"
      />
    </section>
  </section>
</DocumentFragment>
`;
